### PR TITLE
Add org.kiwix.Client application to flathub.

### PR DESCRIPTION
**Pull request by mgautierfr**
The manifest (and the patches) are generated by kiwix-build
(github kiwix/kiwix-build).

Kiwix is a application to allow offline access to content archive (zim),
as Wikipedia, Stackoverflow, ...

The flatpak bundle of the application can already be found at
http://download.kiwix.org/release/kiwix-desktop/org.kiwix.Client.2.0-beta1.flatpak

### CODE REVIEW

**Comment by barthalion** [comment]
testbot, build org.kiwix.Client

File: manifest.json [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    "rename-desktop-file": "kiwix-desktop.desktop",
+    "rename-icon": "kiwix-desktop",
+    "finish-args": [
+        "--socket=wayland",
+        "--socket=x11",
+        "--share=network",
+        "--share=ipc",
+        "--device=dri",
+        "--socket=pulseaudio",
+        "--filesystem=xdg-data"
```
Comment (barthalion) [inline manifest.json]: Does this actually work?

File: manifest.json [hunk 2]
```diff
@@ -0,0 +1,132 @@
+{
+    "app-id": "org.kiwix.Client",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.11",
+    "sdk": "org.kde.Sdk",
+    "command": "kiwix-desktop",
+    "rename-desktop-file": "kiwix-desktop.desktop",
```
Comment (barthalion) [inline manifest.json]: Ideally these should get renamed in upstream repo.

File: manifest.json [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    "url": "http://download.kiwix.org/dev/pugixml-1.2.tar.gz",
+                    "dest": "src"
+                },
+                {
+                    "type": "patch",
+                    "path": "patches/pugixml_meson.patch"
+                }
+            ],
+            "buildsystem": "meson",
+            "builddir": true
```
Comment (barthalion) [inline manifest.json]: `builddir` is already default for meson, also `sources` is usually the last element; `no-autogen` defaults to `false`

File: manifest.json [hunk 4]
```diff
@@ -0,0 +1,132 @@
+{
```
Comment (barthalion) [inline manifest.json]: filename needs to match appid

File: manifest.json (hunk 1 above)
Comment (mgautierfr) [reply manifest.json]: Hum yes. Why it should not ?
Do I missed something ?

File: manifest.json (hunk 2 above)
Comment (mgautierfr) [reply manifest.json]: I'm not sure about this.
The project is named `kiwix-desktop` and it can be installed another way that with flathub.
(We create appimage already and may create "classic" packages)
I prefer to keep a generic name in upstream repository and have flatpak specific name in the flatpak side.

File: manifest.json (hunk 3 above)
Comment (mgautierfr) [reply manifest.json]: Fixed in last forced push.

File: manifest.json (hunk 4 above)
Comment (mgautierfr) [reply manifest.json]: Fixed in last forced push.

File: manifest.json (hunk 1 above)
Comment (barthalion) [reply manifest.json]: I just don't understand it's supposed to achieve. XDG_DATA_DIR env var is overwritten by flatpak and application has access to it by default.

File: manifest.json (hunk 2 above)
Comment (barthalion) [reply manifest.json]: It's mentioned in desktop launcher and appdata specifications that appid form is recommended. It's not flatpak specific.

**Comment by barthalion** [comment]
testbot, build org.kiwix.Client

File: manifest.json (hunk 1 above)
Comment (mgautierfr) [reply manifest.json]: The purpose was to allow the application to access to XDG_DATA_DIR.
But I've just test without it and it still works, good to know.
I will remove it.

File: manifest.json (hunk 2 above)
Comment (mgautierfr) [reply manifest.json]: But we use appid only in flatpak. Everywhere else, the name is `kiwix-desktop`.

**Comment by mgautierfr** [comment]
Last forced push rename the app from `org.kiwix.Client` to `org.kiwix.desktop`. Do I need to create another PR or it is ok ?

File: org.kiwix.desktop.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
+{
+    "app-id": "org.kiwix.desktop",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.11",
```
Comment (barthalion) [inline org.kiwix.desktop.json]: Would be nice to switch to 5.12, it's out since 2 weeks or so.

File: org.kiwix.desktop.json [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            ]
+        },
+        {
+            "name": "kiwix-desktop",
+            "buildsystem": "qmake",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/kiwix/kiwix-desktop.git",
+                    "tag": "2.0-beta2"
```
Comment (barthalion) [inline org.kiwix.desktop.json]: Specific commits are preferred as tags can be modified to point to different commit.

**Comment by barthalion** [comment]
It's okay, branch name is irrelevant for anything but our convenience.

testbot, build org.kiwix.desktop

**Comment by barthalion** [comment]
Looks like appdata file is missing. I see it in [your repo](https://github.com/kiwix/kiwix-desktop/blob/master/resources/org.kiwix.desktop.appdata.xml); should it be installed by build system?

**Comment by barthalion** [comment]
Hey @mgautierfr, any chance you could take a look?

**Comment by mgautierfr** [comment]
Last forced push changed : 
- Use org.kde 5.12 instead of 5.11
- Specify the exact sha of the commit alongside of the tag
- Use last version of the project (correctly export the appdata and the register the mimetype)

File: org.kiwix.desktop.json (hunk 5 above)
Comment (mgautierfr) [reply org.kiwix.desktop.json]: Changed in last forced push.

File: org.kiwix.desktop.json (hunk 6 above)
Comment (mgautierfr) [reply org.kiwix.desktop.json]: Changed in last forced push.

File: manifest.json (hunk 2 above)
Comment (mgautierfr) [reply manifest.json]: We don't rename it anymore. We use the correct "org.kiwix.desktop" name in the upstream project.

**Comment by nedrichards** [comment]
testbot, build org.kiwix.Client
bot, build org.kiwix.Client

**Comment by mgautierfr** [comment]
This is `org.kiwix.desktop` not `org.kiwix.Client`

testbot, build org.kiwix.desktop

**Comment by mgautierfr** [comment]
testbot, build org.kiwix.desktop

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.kiwix.desktop

**Comment by mgautierfr** [comment]
@nedrichards, Please also add @kelson42
